Fix enable/disable bugs

seamusmac opened the pull request:
The full calendar component gets enabled disabled now, not just the input box.
This also fixes isEnabled on MaterialValueBox which was not calling the super setEnabled method.
The waves was messing things up a bit. Removed! 
Does gwt-material really want to force wave style on enable/disable of components?

BenDol commented:
Thanks for the contribution, you will need to squish the commits and replace tabs with 4 spaces to match the code standard. https://github.com/GwtMaterialDesign/gwt-material/wiki/Contributing

seamusmac commented:
@BenDol 
I squashed and cleaned up the commit. Just getting used to using git as well.
Let me know if you need anything, or have any other comments.

kevzlou7979 commented on gwt-material/src/main/java/gwt/material/design/client/base/mixin/EnabledMixin.java:
```diff
@@ -69,21 +67,16 @@ private void applyEnabledProperty(boolean enabled, UIObject obj) {
-                    uiObject.removeStyleName(((HasWaves) uiObject).getWaves().getCssName());
+        }
+        if(obj instanceof HasWidgets) {
+            for(Widget widget: (HasWidgets) obj) {
```
This will break the Tab component , specially the design. Because Link has default disabled styles that will override because you loop the entire Widget then set it to disable.

kevzlou7979 replied:
Just make a test case all the components using Enabled Mixins and see what will happen if you use your commit logic :)

seamusmac commented:
@kevzlou7979  Removed the loop, so it doesn't mess with other components.

seamusmac commented:
Any update/comments on this PR?

kevzlou7979 commented:
We are testing this out :)
